creator-node/src/routes/tracks.js

```diff
@@ -304,8 +304,17 @@ module.exports = function (app) {
         transcodedTrackUUID
       } = req.body
 
+      const prometheusRegistry =
+        req.app.get('serviceRegistry').prometheusRegistry
+      const routePostTracksDurationSecondsMetric =
+        prometheusRegistry.getMetricInstance(
+          prometheusRegistry.metricNames.ROUTE_POST_TRACKS_DURATION_SECONDS_HISTOGRAM
+        )
+      const metricEndTimerFn = routePostTracksDurationSecondsMetric.startTimer()
+
       // Input validation
       if (!blockchainTrackId || !blockNumber || !metadataFileUUID) {
+        metricEndTimerFn({ code: 400 })
```

Comment:
what happens if you forget to not call the callback? probably nothing right? it just doesn't track that metric?

Reply:
reason i'm asking is this is scattered in a lot of places but idk if we can really make this better

Reply:
yep exactly - it does nothing, when you call `histogram.startTimer()`, it literally just creates and returns a function with internal state for `startTime`
if you don't call that second function the state of `histogram` metric instance is never changed i.e. the internal `observe()` call is never made

Reply:
yeh idk, this seems like basically the established pattern in prometheus, not sure how to do better, or if worth doing

the one concern i had that your question raised was if you call `metric.startTimer()` in one place and then call `metric.startTimer()` elsewhere e.g. if single metric is consumed in multiple places, if that would corrupt state
as the above code snippet shows tho, this is not the case. the `metric` instance's state is only changed when the `endTimer()` function is called